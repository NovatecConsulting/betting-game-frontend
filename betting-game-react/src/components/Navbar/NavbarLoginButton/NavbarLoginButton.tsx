import React from "react";
import { useLocation } from "react-router-dom";
import { AuthorizationManager } from "../../../auth";
import { ROUTE_LOGIN_CALLBACK } from "../../../routes";

// TODO: build button component
const NavbarLoginButton: React.FC = () => {
    const location = useLocation();
    const user = AuthorizationManager.getUser();
    return (
        <div className="float-right">
            {user?.profile ? (
                <>
                    {user.profile.picture && (
                        <img
                            src={user.profile.picture}
                            alt={user.profile.email}
                            width={25}
                            height={25}
                        />
                    )}
                    <div className="text-white">{user.profile.email}</div>
                    <div
                        onClick={() => AuthorizationManager.logout()}
                        className="cursor-pointer inline-block text-sm px-5 py-2 lg:mt-0 mr-10 leading-none border rounded-full text-white border-white hover:text-blue-400 hover:bg-white active:bg-blue-700"
                    >
                        Logout
                    </div>
                </>
            ) : location.pathname === ROUTE_LOGIN_CALLBACK ? null : (
                <div
                    onClick={() => AuthorizationManager.login()}
                    className="cursor-pointer inline-block text-sm px-5 py-2 lg:mt-0 mr-10 leading-none border rounded-full text-white border-white hover:text-blue-400 hover:bg-white active:bg-blue-700"
                    data-testid="Login-NavButton"
                >
                    <span data-testid="Login-NavButtonText">Login</span>
                </div>
            )}
        </div>
    );
};

export default NavbarLoginButton;
