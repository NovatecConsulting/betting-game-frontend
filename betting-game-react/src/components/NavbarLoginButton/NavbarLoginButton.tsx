import React from 'react';
import { NavLink } from 'react-router-dom';

import { LOGIN } from '../../config/routes';

const NavbarLoginButton: React.FC = () => (
  <div className='float-right'>
    <NavLink
      to={LOGIN.path}
      className='inline-block text-sm px-5 py-2 lg:mt-0 mr-10 leading-none border rounded-full text-white border-white hover:text-blue-400 hover:bg-white active:bg-blue-700'
      data-testid='Login-NavButton'
    >
      <span data-testid='Login-NavButtonText'>Login</span>
    </NavLink>
  </div>
);

export default NavbarLoginButton;
