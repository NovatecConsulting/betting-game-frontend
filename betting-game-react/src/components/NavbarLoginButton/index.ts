export { default as NavbarLoginButton } from "./NavbarLoginButton";
