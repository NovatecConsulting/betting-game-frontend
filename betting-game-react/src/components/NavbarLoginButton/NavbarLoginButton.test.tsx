import '@testing-library/jest-dom/extend-expect';

import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { render } from '@testing-library/react';

import NavbarLoginButton from './NavbarLoginButton';

describe('Login button for the navigation bar', () => {
  test('it should render a non-empty button with text', () => {
    const { getByTestId } = render(
      <Router>
        <NavbarLoginButton />
      </Router>
    );

    const navbarLoginButton = getByTestId('Login-NavButton');
    const navbarLoginButtonText = getByTestId('Login-NavButtonText');

    expect(navbarLoginButton).toBeInTheDocument();
    expect(navbarLoginButtonText).not.toBeEmpty();
  });
});
