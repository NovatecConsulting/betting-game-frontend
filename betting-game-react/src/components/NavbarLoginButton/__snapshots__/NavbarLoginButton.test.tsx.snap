// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLoginButton component should render and match snapshot with correct content. 1`] = `
<div
  className="float-right"
>
  <NavLink
    className="inline-block text-sm px-5 py-2 lg:mt-0 mr-10 leading-none border rounded-full text-white border-white hover:text-blue-400 hover:bg-white active:bg-blue-700"
    data-testid="Login-NavButton"
    to="/login"
  >
    <span
      data-testid="Login-NavButtonText"
    >
      Login
    </span>
  </NavLink>
</div>
`;
